pkg/maps/ipmasq/ipmasq.go

```diff
@@ -86,32 +86,42 @@ type IPMasqBPFMap struct{}
 
 func (*IPMasqBPFMap) Update(cidr net.IPNet) error {
 	if ip.IsIPv4(cidr.IP) {
```

Comment:
Could combine the if and the one below and return if ipv4 is enable. No need for else clause

Reply:
Couldn't?

I understand you mean the following:

```go
	if ip.IsIPv4(cidr.IP) && option.Config.EnableIPv4Masquerade {
		return IPMasq4Map().Update(keyIPv4(cidr), &Value{})
	}
	if option.Config.EnableIPv6Masquerade {
		return IPMasq6Map().Update(keyIPv6(cidr), &Value{})
	}
	return nil
```

What if the CIDR is IPv4, `EnableIPv4Masquerade` is not set, but `EnableIPv6Masquerade` is set? In this case, without the `else`, we would attempt to add the CIDR to the IPv6 map, instead of skipping the entry.

Or did I get you wrong?

Reply:
Rats! I see what you are saying. You are totally correct!